appeals/web/src/server/appeals/appeal-details/appeal-timetables/appeal-timetables.mapper.js

```diff
@@ -0,0 +1,114 @@
+};
+
+/**
+ * @param {import('./appeal-timetables.service.js').AppealTimetables} appealTimetables
+ * @param { 'finalCommentReviewDate' | 'issueDeterminationDate' | 'lpaQuestionnaireDueDate' | 'statementReviewDate' } timetableProperty
+ * @returns {AppealTimetablesMap}
+ */
+export const appealTimetablesMapper = (appealTimetables, timetableProperty) => {
+	const currentDueDateIso = appealTimetables && appealTimetables[timetableProperty];
+	const currentDueDate = currentDueDateIso && dateToDisplayDate(currentDueDateIso);
+	const changeOrScheduleText = currentDueDate ? 'Change' : 'Schedule';
+
+	const map = {
+		finalCommentReviewDate: {
+			sideNote:
+				currentDueDate && `The current due date for the final comment review is ${currentDueDate}`,
+			page: {
+				title: 'Final comment review due date',
+				text: `${changeOrScheduleText} final comment review due date`
+			},
+			confirmation: {
+				title: 'Final comment review due date updated',
+				preTitle: 'The due date for the final comment review due date has been updated.',
+				rows: [
+					{
+						text: 'We’ve sent an email to the appellant to inform them about changes to the timetable.'
+					}
+				]
+			}
+		},
+		issueDeterminationDate: {
+			sideNote:
+				currentDueDate && `The current due date for the issue determination is ${currentDueDate}`,
+			page: {
+				title: 'Issue determination due date',
+				text: `${changeOrScheduleText} issue determination due date`
+			},
+			confirmation: {
+				title: 'Issue determination due date updated',
+				preTitle: 'The due date for the issue determination has been updated.',
+				rows: [
+					{
+						text: 'We’ve sent an email to the appellant to inform them about changes to the timetable.'
+					}
+				]
+			}
+		},
+		lpaQuestionnaireDueDate: {
+			sideNote:
+				currentDueDate && `The current due date for the LPA questionnaire is ${currentDueDate}`,
+			page: {
+				title: 'LPA questionnaire due date',
+				text: `${changeOrScheduleText} LPA questionnaire due date`
+			},
+			confirmation: {
+				title: 'LPA Questionnaire due date updated',
+				preTitle: 'The due date for the LPA Questionnaire has been updated.',
+				rows: [
+					{
+						text: 'We’ve sent an email to the appellant and LPA to inform them about changes to the timetable.'
+					},
+					{
+						text: 'We also sent them a reminder about the appeal’s due date.'
+					}
+				]
+			}
+		},
+		statementReviewDate: {
+			sideNote:
+				currentDueDate && `The current due date for the statement review is ${currentDueDate}`,
+			page: {
+				title: 'Statement review due date',
+				text: `${changeOrScheduleText} statement review due date`
+			},
+			confirmation: {
+				title: 'Statement review due date updated',
+				preTitle: 'The due date for the statement review has been updated.',
+				rows: [
+					{
+						text: 'We’ve sent an email to the appellant to inform them about changes to the timetable.'
+					}
+				]
+			}
+		}
+	};
+
+	return map[timetableProperty];
```

Comment:
1. On the point of avoiding the need for `routeToObjectMapper` it is useful to have define the map once because it is used in multiple places. For example this map is used for patch request when updating timetable

2. On the point of "wastefulness": the JIRA ticket has only one example. When it comes to testing I anticipate there might be differences in Due Date page content, depending what for you are submitting the due date. For example, right now I have 2 rows for LPAQ and only 1 for everything else. I do not see how this is wasteful as you need to return only one object for the page type you are requesting

Reply:
1. I see what you mean about needing `routeToObjectMapper` for the patch request, I'd missed that. In `processUpdateDueDate` you're using it to map the updated timetable data to the API patch request, but that should probably be done in the mapper (it would just be a case of creating a new function in the mapper file). Then you could avoid having any references to `routeToObjectMapper` outside the mapper file, and instead pass in `timetableType` to the relevant mapper functions. I just feel it's best to keep all mapping-related logic inside the mapper file for separation of concerns.
2. Maybe wasteful was a strong word - I just feel this function is a lot more verbose than it needs to be. Most of the data structure is the same for each item (`finalCommentReviewDate`, `lpaQuestionnaireDueDate` etc.) - this could be a lot shorter if you build up the object conditionally based on what's passed in to the function. For example: 
```export const appealTimetablesMapper = (appealTimetables, timetableProperty) => {
	const currentDueDateIso = appealTimetables && appealTimetables[timetableProperty];
	const currentDueDate = currentDueDateIso && dateToDisplayDate(currentDueDateIso);
	const changeOrScheduleText = currentDueDate ? 'Change' : 'Schedule';

	let categoryText = '';
	const rows = [
		{
			text: 'We’ve sent an email to the appellant to inform them about changes to the timetable.'
		}
	];

	switch (timetableProperty) {
		case 'finalCommentReviewDate':
			categoryText = 'final comment review';
			break;
		case 'issueDeterminationDate':
			categoryText = 'issue determination';
			break;
		case 'lpaQuestionnaireDueDate':
			categoryText = 'LPA questionnaire';
			rows.push({
				text: 'We also sent them a reminder about the appeal’s due date.'
			});
			break;
		default:
			break;
	}

	return {
		sideNote:
			currentDueDate && `The current due date for the ${categoryText} is ${currentDueDate}`,
		page: {
			title: `${categoryText.toUpperCase()} due date`,
			text: `${changeOrScheduleText} final comment review due date`
		},
		confirmation: {
			title: `${categoryText.toUpperCase()} due date updated`,
			preTitle: `The due date for the ${categoryText} has been updated.`,
			rows
		}
	};
}```

Reply:
refactored. done